client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -498,11 +517,15 @@ public void write(PendingEvent event) {
             try {
                 // if connection is null getConnection() establishes a connection and retransmits all events in inflight
                 // list.
-                connection = Futures.getThrowingException(getConnection());
+                connection = Futures.getThrowingExceptionWithTimeout(getConnection(), clientConfig.getConnectTimeoutMilliSec());
```

Comment:
So, looking at this there already is a timeout of sorts on line 248. The problem is the internal thread can be blocked on the getInfo without time. This means the future on 673 would never return. So if this timeout was introduced all requests would simply timeout here and the internal thread would never unblock and the reader would never work.

This seemed familiar, and I looked, and in fact we made this change before: https://github.com/pravega/pravega/pull/6528/files#diff-77e930ef537f025820cd9af3e14f393bef92097b9f7978e7f3b57bfc3f28a1ab
That change ended up being rolled back due to connection churn, but I think it's approach to this file is the right approach. For now, let's copy that approach to this file except the change to the `reconnect` call below, for that one I think we'll need to have a meeting to discuss, as `reconnect` is definitely not right, but it did cause problems to change it.

Reply:
will do this in #7194